refactor(account-management)_: simplify account management and enhance functionality

saledjenic opened the pull request:
This one brings a lot of clarity for both the account generator and the account manager. More about changes can be found in each commit's description. The first commit is about generator improvements, the second about account manager improvements, and the third is about fixing the tests and aligning the rest of the app with the changes done in the previous two commits.

----
Reference to other accounts-management refactor PRs:
- https://github.com/status-im/status-go/pull/6605
- https://github.com/status-im/status-go/pull/6612
- https://github.com/status-im/status-go/pull/6640
- https://github.com/status-im/status-go/pull/6660
- https://github.com/status-im/status-go/pull/6668
- https://github.com/status-im/status-go/pull/6673
- https://github.com/status-im/status-go/pull/6686
- https://github.com/status-im/status-go/pull/6687
- https://github.com/status-im/status-go/pull/6692

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (77)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 02822795 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/1/) | 2025-06-12 12:53:28 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250612-125037-028227-pr6668.aar) |
| :heavy_check_mark: | 02822795 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/1/) | 2025-06-12 12:53:41 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-125042-028227-pr6668.zip) |
| :heavy_check_mark: | 02822795 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/1/) | 2025-06-12 12:54:11 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250612-125056-028227-pr6668.zip) |
| :heavy_check_mark: | 02822795 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/1/) | 2025-06-12 12:55:35 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250612-125102-028227-pr6668.zip) |
| :heavy_check_mark: | 02822795 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/2/) | 2025-06-12 12:55:49 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250612-125335-028227-pr6668.aar) |
| :heavy_check_mark: | 02822795 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/2/) | 2025-06-12 12:56:24 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-125344-028227-pr6668.zip) |
| :heavy_check_mark: | 02822795 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/1/) | 2025-06-12 12:56:38 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250612-125102-028227-pr6668.zip) |
| :heavy_check_mark: | 02822795 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/1/) | 2025-06-12 12:58:49 | ~8 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250612-125042-028227-pr6668.zip) |
| :heavy_check_mark: | 02822795 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/1/) | 2025-06-12 13:01:09 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-125042-028227-pr6668.zip) |
| :heavy_check_mark: | 02822795 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/2/) | 2025-06-12 13:05:52 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250612-125900-028227-pr6668.zip) |
| :heavy_check_mark: | 02822795 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/2/) | 2025-06-12 13:08:45 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-130114-028227-pr6668.zip) |
| :heavy_multiplication_x: | 02822795 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/1/) | 2025-06-12 13:08:58 | ~17 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/1//consoleText) |
| :heavy_multiplication_x: | 02822795 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/1/) | 2025-06-12 13:17:16 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 6a9e0e64 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/3/) | 2025-06-12 19:26:59 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250612-192424-6a9e0e-pr6668.aar) |
| :heavy_check_mark: | 6a9e0e64 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/2/) | 2025-06-12 19:27:39 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250612-192429-6a9e0e-pr6668.zip) |
| :heavy_check_mark: | 6a9e0e64 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/3/) | 2025-06-12 19:27:46 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-192426-6a9e0e-pr6668.zip) |
| :heavy_check_mark: | 6a9e0e64 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/2/) | 2025-06-12 19:29:06 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250612-192435-6a9e0e-pr6668.zip) |
| :heavy_check_mark: | 6a9e0e64 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/2/) | 2025-06-12 19:30:04 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250612-192431-6a9e0e-pr6668.zip) |
| :heavy_check_mark: | 6a9e0e64 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/3/) | 2025-06-12 19:30:19 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250612-192425-6a9e0e-pr6668.zip) |
| :heavy_check_mark: | 6a9e0e64 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/3/) | 2025-06-12 19:35:00 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-192427-6a9e0e-pr6668.zip) |
| :heavy_multiplication_x: | 6a9e0e64 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/2/) | 2025-06-12 19:36:14 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/2//consoleText) |
| :heavy_multiplication_x: | 6a9e0e64 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/2/) | 2025-06-12 19:49:23 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4b384731 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/4/) | 2025-06-12 19:35:41 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250612-193317-4b3847-pr6668.aar) |
| :heavy_check_mark: | 4b384731 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/4/) | 2025-06-12 19:36:26 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-193319-4b3847-pr6668.zip) |
| :heavy_check_mark: | 4b384731 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/3/) | 2025-06-12 19:36:28 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250612-193322-4b3847-pr6668.zip) |
| :heavy_check_mark: | 4b384731 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/3/) | 2025-06-12 19:38:03 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250612-193328-4b3847-pr6668.zip) |
| :heavy_check_mark: | 4b384731 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/3/) | 2025-06-12 19:38:51 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250612-193324-4b3847-pr6668.zip) |
| :heavy_check_mark: | 4b384731 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/4/) | 2025-06-12 19:38:58 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250612-193318-4b3847-pr6668.zip) |
| :heavy_check_mark: | 4b384731 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/4/) | 2025-06-12 19:43:27 | ~8 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-193504-4b3847-pr6668.zip) |
| :heavy_check_mark: | 4b384731 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/3/) | 2025-06-12 19:48:13 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/3//consoleText) |
| :heavy_multiplication_x: | 4b384731 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/3/) | 2025-06-12 20:14:06 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 33b38ff5 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/5/) | 2025-06-13 08:00:13 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250613-075701-33b38f-pr6668.aar) |
| :heavy_check_mark: | 33b38ff5 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/4/) | 2025-06-13 08:00:19 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250613-075707-33b38f-pr6668.zip) |
| :heavy_check_mark: | 33b38ff5 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/5/) | 2025-06-13 08:00:31 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250613-075703-33b38f-pr6668.zip) |
| :heavy_check_mark: | 33b38ff5 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/4/) | 2025-06-13 08:01:47 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250613-075712-33b38f-pr6668.zip) |
| :heavy_check_mark: | 33b38ff5 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/4/) | 2025-06-13 08:02:48 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250613-075709-33b38f-pr6668.zip) |
| :heavy_check_mark: | 33b38ff5 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/5/) | 2025-06-13 08:04:10 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250613-075702-33b38f-pr6668.zip) |
| :heavy_check_mark: | 33b38ff5 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/5/) | 2025-06-13 08:07:22 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250613-075704-33b38f-pr6668.zip) |
| :heavy_multiplication_x: | 33b38ff5 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/4/) | 2025-06-13 08:11:26 | ~14 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/4//consoleText) |
| :heavy_check_mark: | 33b38ff5 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/4/) | 2025-06-13 08:22:23 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/4//consoleText) |
| :heavy_multiplication_x: | 33b38ff5 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/5/) | 2025-06-13 09:02:59 | ~17 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 59c1c7d3 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/6/) | 2025-06-16 11:26:50 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250616-112407-59c1c7-pr6668.aar) |
| :heavy_check_mark: | 59c1c7d3 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/6/) | 2025-06-16 11:27:01 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-112408-59c1c7-pr6668.zip) |
| :heavy_check_mark: | 59c1c7d3 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/5/) | 2025-06-16 11:27:23 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-112412-59c1c7-pr6668.zip) |
| :heavy_check_mark: | 59c1c7d3 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/5/) | 2025-06-16 11:28:53 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250616-112418-59c1c7-pr6668.zip) |
| :heavy_check_mark: | 59c1c7d3 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/5/) | 2025-06-16 11:29:49 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-112415-59c1c7-pr6668.zip) |
| :heavy_check_mark: | 59c1c7d3 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/6/) | 2025-06-16 11:30:19 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250616-112408-59c1c7-pr6668.zip) |
| :heavy_multiplication_x: | 59c1c7d3 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/6/) | 2025-06-16 11:31:30 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/6//consoleText) |
| :heavy_check_mark: | 59c1c7d3 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/6/) | 2025-06-16 11:35:12 | ~11 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-112410-59c1c7-pr6668.zip) |
| | | | | | | |
| :heavy_check_mark: | 376d831d | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/7/) | 2025-06-16 11:29:57 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250616-112657-376d83-pr6668.aar) |
| :heavy_check_mark: | 376d831d | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/7/) | 2025-06-16 11:30:27 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-112705-376d83-pr6668.zip) |
| :heavy_check_mark: | 376d831d | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/6/) | 2025-06-16 11:31:46 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-112728-376d83-pr6668.zip) |
| :heavy_check_mark: | 376d831d | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/6/) | 2025-06-16 11:33:34 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250616-112856-376d83-pr6668.zip) |
| :heavy_check_mark: | 376d831d | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/6/) | 2025-06-16 11:35:31 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-112954-376d83-pr6668.zip) |
| :heavy_check_mark: | 376d831d | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/7/) | 2025-06-16 11:36:56 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250616-113037-376d83-pr6668.zip) |
| :heavy_check_mark: | 376d831d | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/7/) | 2025-06-16 11:38:49 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/7//consoleText) |
| :heavy_check_mark: | 376d831d | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/7/) | 2025-06-16 11:44:07 | ~8 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-113523-376d83-pr6668.zip) |
| :heavy_check_mark: | 376d831d | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/6/) | 2025-06-16 11:52:35 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 001e3518 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/8/) | 2025-06-16 11:45:25 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250616-114244-001e35-pr6668.aar) |
| :heavy_check_mark: | 001e3518 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/7/) | 2025-06-16 11:46:03 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-114249-001e35-pr6668.zip) |
| :heavy_check_mark: | 001e3518 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/8/) | 2025-06-16 11:47:27 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-114246-001e35-pr6668.zip) |
| :heavy_check_mark: | 001e3518 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/7/) | 2025-06-16 11:47:33 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250616-114257-001e35-pr6668.zip) |
| :heavy_check_mark: | 001e3518 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/7/) | 2025-06-16 11:48:31 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-114253-001e35-pr6668.zip) |
| :heavy_check_mark: | 001e3518 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/8/) | 2025-06-16 11:48:51 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250616-114245-001e35-pr6668.zip) |
| :heavy_multiplication_x: | 001e3518 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/8/) | 2025-06-16 11:51:53 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/8//consoleText) |
| :heavy_check_mark: | 001e3518 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/8/) | 2025-06-16 11:53:46 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-114416-001e35-pr6668.zip) |
| :heavy_check_mark: | 001e3518 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/7/) | 2025-06-16 12:17:12 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/7//consoleText) |
| :heavy_check_mark: | 001e3518 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/9/) | 2025-06-16 12:31:40 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e4b0ab0a | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/9/) | 2025-06-17 09:59:17 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250617-095632-e4b0ab-pr6668.aar) |
| :heavy_check_mark: | e4b0ab0a | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/9/) | 2025-06-17 10:00:17 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-095635-e4b0ab-pr6668.zip) |
| :heavy_multiplication_x: | e4b0ab0a | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/8/) | 2025-06-17 10:00:24 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/8//consoleText) |
| :heavy_check_mark: | e4b0ab0a | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/8/) | 2025-06-17 10:00:24 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-095638-e4b0ab-pr6668.zip) |
| :heavy_check_mark: | e4b0ab0a | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/8/) | 2025-06-17 10:01:25 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250617-095645-e4b0ab-pr6668.zip) |
| :heavy_check_mark: | e4b0ab0a | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/8/) | 2025-06-17 10:02:29 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-095640-e4b0ab-pr6668.zip) |
| :heavy_check_mark: | e4b0ab0a | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/9/) | 2025-06-17 10:05:07 | ~8 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250617-095634-e4b0ab-pr6668.zip) |
| :heavy_multiplication_x: | e4b0ab0a | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/10/) | 2025-06-17 10:05:19 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/10//consoleText) |
| :heavy_check_mark: | e4b0ab0a | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/9/) | 2025-06-17 10:08:55 | ~12 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-095635-e4b0ab-pr6668.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 72bc140c | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/10/) | 2025-06-17 11:29:18 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250617-112607-72bc14-pr6668.aar) |
| :heavy_check_mark: | 72bc140c | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/10/) | 2025-06-17 11:29:56 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-112610-72bc14-pr6668.zip) |
| :heavy_check_mark: | 72bc140c | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/9/) | 2025-06-17 11:30:06 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-112613-72bc14-pr6668.zip) |
| :heavy_check_mark: | 72bc140c | [#9](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/9/) | 2025-06-17 11:31:13 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250617-112621-72bc14-pr6668.zip) |
| :heavy_check_mark: | 72bc140c | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/10/) | 2025-06-17 11:32:23 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250617-112608-72bc14-pr6668.zip) |
| :heavy_check_mark: | 72bc140c | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/9/) | 2025-06-17 11:32:25 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-112617-72bc14-pr6668.zip) |
| :heavy_multiplication_x: | 72bc140c | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/11/) | 2025-06-17 11:34:46 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/11//consoleText) |
| :heavy_check_mark: | 72bc140c | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/10/) | 2025-06-17 11:38:30 | ~12 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-112611-72bc14-pr6668.zip) |
| :heavy_check_mark: | 72bc140c | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/9/) | 2025-06-17 11:52:35 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6668/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c5b1d026 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6668/11/) | 2025-06-17 12:16:13 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250617-121315-c5b1d0-pr6668.aar) |
| :heavy_check_mark: | c5b1d026 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6668/11/) | 2025-06-17 12:16:42 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-121317-c5b1d0-pr6668.zip) |
| :heavy_check_mark: | c5b1d026 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6668/10/) | 2025-06-17 12:16:49 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-121322-c5b1d0-pr6668.zip) |
| :heavy_check_mark: | c5b1d026 | [#10](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6668/10/) | 2025-06-17 12:18:28 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250617-121328-c5b1d0-pr6668.zip) |
| :heavy_check_mark: | c5b1d026 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6668/10/) | 2025-06-17 12:19:33 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-121324-c5b1d0-pr6668.zip) |
| :heavy_check_mark: | c5b1d026 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6668/11/) | 2025-06-17 12:20:55 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250617-121316-c5b1d0-pr6668.zip) |
| :heavy_check_mark: | c5b1d026 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/12/) | 2025-06-17 12:22:41 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6668/12//consoleText) |
| :heavy_check_mark: | c5b1d026 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6668/11/) | 2025-06-17 12:26:09 | ~12 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-121318-c5b1d0-pr6668.zip) |

saledjenic committed:
refactor(account-generator)_: simplify account management and enhance functionality

- Removed the errors.go file, consolidating error handling within the generator package.
- Refactored account creation methods to improve clarity and usability, introducing CreateAccountFromMnemonic, CreateAccountFromPrivateKey, and CreateAccountFromKey functions.
- Updated tests to align with the new account structure and ensure coverage.
- Enhanced README documentation to reflect the new account creation and derivation methods.

saledjenic committed:
refactor(account-management)_: decouple and enhance keystore functionality

- Removed the old GethManager and associated methods, moving to a new adapter-based approach for keystore management.
- Previously Introduced keystore adapter, updated, improved and moved from internal.
- Updated account management methods to utilize the new adapter, enhancing clarity and usability.
- Refactored tests to align with the new structure and ensure coverage.
- Cleaned up unused code and improved overall readability of the account package.

saledjenic committed:
refactor(account-management)_: enhance keystore integration and account handling

- Updated tests to utilize the new account management.
- Refactored account loading and creation methods to align with the new structure, enhancing maintainability.
- Cleaned up unused code and improved overall readability in the account package.
- This refactor further decouples the account management from direct dependencies on the go-ethereum keystore.

igor-sirotin reviewed (approved):
Didn't read the changes. I see no point in reviewing 2000 lines of code without context.

saledjenic commented:
> Didn't read the changes. I see no point in reviewing 2000 lines of code without context.

@igor-sirotin I see no point in such a comment since this refactor, having in mind that almost all parts of the app use the "old" account management, is not that easy to refactor in small steps, because every change has many implications. We cannot expect that the refactoring of this big part will be a small change.

I know and understand the difficulties of reviewing such a change, and sorry about that, but any other approach would take significantly longer and would require going over the same parts again and again. 

Since the entire refactor work will be done via several PRs, my suggestion would be to have everything merged in the base branch `modules/accounts-management-3` (which is this PR https://github.com/status-im/status-go/pull/6640) then everybody will be able to see all the changes there before merging. Also since the change improves code a lot, removing redundant stuff, makes it more maintainable, clear, explicit and reusable, what's really important is to not break anything, while if I forgot to put some number as const or could name some function better is less important, and all that can be done once the change gets merged. We can improve more once the approach is changed. What I am doing with this refactor is defining the scope of responsibilities of account management. 

Also, as I already promised in some comment, I will try to prepare some high-level class diagram these days to clarify the structure of the new account management part.

saledjenic committed:
refactor(account-management)_: remove onboarding functionality and related tests

- Deleted the onboarding-related methods and tests from the account package.
- Cleaned up unused code.
- Adjusted error handling by removing onboarding-specific errors.

osmaczko commented on protocol/communities_events_token_master_test.go:
```diff
@@ -219,6 +219,8 @@ func (s *TokenMasterCommunityEventsSuite) TestReceiveRequestsToJoinWithRevealedA
 }
 
 func (s *TokenMasterCommunityEventsSuite) TestTokenMasterAcceptsRequestToJoinAfterMemberLeave() {
+	s.T().Skip("flaky test")
```
Tests are re-run in the CI, so flaky tests shouldn't be blocking (i.e., they shouldn't cause the test job to turn red). Did this test fail on every re-run?

saledjenic replied:
The test fails only if the entire suite is run, if the test is run individually, all is good. Because of that and cause I am not familiar with the tests there, I assume that some of the tests that are run before that one have changed the local state that somehow popped up that test to fail. It would be really nice if someone from the core team could have a look. By the way, there is already one more test marked as flaky there.

osmaczko replied:
I reverted the change and tests are green https://github.com/status-im/status-go/pull/6674. I think it is safe to remove this skip.

osmaczko commented on protocol/messenger_config.go:
```diff
@@ -72,6 +73,11 @@ type MessengerSignalsHandler interface {
 	SendCuratedCommunitiesUpdate(response *communities.KnownCommunitiesResponse)
 }
 
+type Manager interface {
```
```suggestion
type AccountManager interface {
```

saledjenic replied:
It's handled in this PR https://github.com/status-im/status-go/pull/6687
